PlayListMaker. Домашнее задание. Спринт 8.

AlexanderKorytin opened the pull request.

AlexanderKorytin committed:
сделана верска главного экрана и экрана настройки

AlexanderKorytin committed:
реализованны слушатели для трех кнопок главного меню:
1. Поиск - с помощью анонимного класса с заглушкой Toast
2. Медиатека - с помощью лямбда выражения с заглушкой Toast
3. Настройки - с помощью лямбда выражения с заглушкой Toast

AlexanderKorytin committed:
Создан класс SettingsActivity.
Объявлен в AndroidManifest.xml.
Установлена для SettingsActivity, созданная в предыдущей теме разметка.
Созданы две пустых Activity для медиатеки и для поиска.
Объявлены в AndroidManifest.xml.
Убран код Toast из обработчиков нажатия на кнопки.
Добавлен код переходов на экраны в соответствующие обработчики.
Для кнопки «Назад» экрана «Настройки» добавлен обработчик нажатий. Добавлен код перехода на главный экран в обработчик.

AlexanderKorytin committed:
умеличена площадь нажатия кнопки "назад" меню Settings и скорректирован ее обработчик на другой View

AlexanderKorytin committed:
Изменены цвета фонов меню

ilshat-abdulin commented on app/src/main/java/com/example/playlistmaker/MainActivity.kt:
```diff
@@ -1,11 +1,41 @@
+        buttonMedia.setOnClickListener {
+            val mediaIntent = Intent(this, MediaActivity::class.java)
+            startActivity(mediaIntent)
+            // Toast.makeText(this@MainActivity, "Нажали кнопку Медиатека!", Toast.LENGTH_SHORT).show()
```
🍏 Лучше неиспользуемый код просто удалить

ilshat-abdulin commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -0,0 +1,22 @@
+        val back = findViewById<LinearLayout>(R.id.backSetting)
+        back.setOnClickListener {
+            val backIntent = Intent(this, MainActivity::class.java)
+            startActivity(backIntent)
```
🍏 Правильным способом для навигации в предыдущий экран, будет использование метода `finish()`, который при вызове скроет и завершит текущую активити и выполнится навигация в предыдущую активити, которая хранится в памяти. А вызов `startActivity` лишь создаст еще один экземпляр `MainActivity` и стек навигации будет выглядеть так: 
`MainActivity->SettingsActivity->MainActivity`. В этом можно убедиться, нажав системную кнопку "Назад"

```
back.setOnClickListener {
    finish()
}
```

ilshat-abdulin commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,79 @@
+        app:cornerRadius="16dp"
+        app:icon="@drawable/find"
+        app:iconGravity="textStart"
+        app:iconPadding="@dimen/icon_indentation"
```
🍏 Лучше привыкать для экспорта иконки, выбирать самый верхний слой. Отмечен красным
![image](https://github.com/AlexanderKorytin/PlaylistMaker/assets/72722086/11a307f5-1503-403b-9e6d-3ea953f8d251)

И отступ нужно учесть при таком размере
![image](https://github.com/AlexanderKorytin/PlaylistMaker/assets/72722086/6a20661e-c745-4118-9ff1-fe8a752af1ad)

ilshat-abdulin commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,113 @@
+            android:layout_marginTop="12dp"
+            android:text="@string/bot_settings"
+            android:textSize="@dimen/primary_text_size" />
+    </LinearLayout>
```
⚠ Отступ для заголовка, слева и сверху, отличается 

Как должно быть по дизайну
![image](https://github.com/AlexanderKorytin/PlaylistMaker/assets/72722086/792fe924-923d-4ba3-9476-ef7d1e397ee9)
![image](https://github.com/AlexanderKorytin/PlaylistMaker/assets/72722086/28baa8f2-d531-40f5-9006-bd4f4f06da31)

И слишком большой отступ снизу до переключателя. 
![image](https://github.com/AlexanderKorytin/PlaylistMaker/assets/72722086/3357c8b7-a009-4eba-a5cf-81457cd0328f)


В целом, имеется излишняя вложенность из LinearLayout, которую можно решить так. 
```
    <LinearLayout
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:orientation="horizontal">

        <ImageView
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_marginStart="@dimen/padding_20"
            android:layout_marginVertical="20dp"
            android:foreground="?selectableItemBackground"
            android:src="@drawable/back"
            tools:ignore="RtlSymmetry" />

        <TextView
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_gravity="center"
            android:layout_marginStart="28dp"
            android:text="@string/bot_settings"
            android:textSize="@dimen/primary_text_size" />
    </LinearLayout>
```

Отступы берешь, выбрав слой Panel Header
![image](https://github.com/AlexanderKorytin/PlaylistMaker/assets/72722086/9304e2ad-0573-4704-b1a7-11e9d02aacda)
![image](https://github.com/AlexanderKorytin/PlaylistMaker/assets/72722086/bb2c6a31-e2eb-4ca9-806d-934e4b0ffdff)

ilshat-abdulin commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,113 @@
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/zero"
+        android:layout_weight="2"
```
⚠ Лучше вес тут и в остальных компонентах не применять. Нет необходимости, чтобы компоненты в каких-то пропорциях располагались на экране. 

И можно обойтись без этого LinearLayout, если везде уберешь вес. Можно просто соблюдать отступы, располагать компоненты поочередно и всё встанет ровно.

ilshat-abdulin commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,113 @@
+            android:paddingLeft="@dimen/padding_16"
+            android:text="@string/dark_theme"
+            android:textSize="@dimen/set_text_size"
+            tools:ignore="UseSwitchCompatOrMaterialXml" />
```
Если убрать везде android:layout_weight="1" и избавится от лишней вложенности, то этот Switch можно расположить, сразу после заголовка, таким образом

```
        <Switch
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:layout_marginTop="24dp"
        android:paddingVertical="21dp"
        android:paddingStart="@dimen/padding_16"
        android:text="@string/dark_theme"
        android:textSize="@dimen/set_text_size"
        tools:ignore="UseSwitchCompatOrMaterialXml" />
```

Просто ориентируясь на отступы 
![image](https://github.com/AlexanderKorytin/PlaylistMaker/assets/72722086/0b4f3f4b-42ad-4d9f-a411-ce9468c3c906)

![image](https://github.com/AlexanderKorytin/PlaylistMaker/assets/72722086/0f717aac-5994-4353-bb53-8017f8675868)

ilshat-abdulin commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,113 @@
+            android:textColor="@color/black"
+            android:textSize="@dimen/set_text_size"
+            android:typeface="sans"
+            app:icon="@drawable/share"
```
⚠ Видно, как выделяются кнопки, хотя по дизайну все в одной плоскости и нет намека на то, что нужно использовать Button. Кнопку нужно использовать, когда она явно нужна.

![image](https://github.com/AlexanderKorytin/PlaylistMaker/assets/72722086/81c31704-c7bd-4704-ac0a-33a4e59dcd50)


Можно по аналогии с header использовать LinearLayout, в котором расположить текст и иконку. Или более хороший вариант, использовать TextView c атрибутом app:drawableEndCompat, смысл тот же, как у app:icon у кнопок. 

И соблюсти отступы, по аналогии с Switch
![image](https://github.com/AlexanderKorytin/PlaylistMaker/assets/72722086/fd0399ec-c38d-44c2-bb4a-c805c18a9984)

Обязательно экспортируй иконку выбрав верхний слой, т.е. с размером 24х24
![image](https://github.com/AlexanderKorytin/PlaylistMaker/assets/72722086/ab76ddee-d31f-4224-bcd1-329751238527)

И примерно так должно получится. Пользуйся)

```
    <TextView
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:paddingVertical="21dp"
        android:paddingStart="@dimen/padding_16"
        android:paddingEnd="12dp"
        android:text="@string/share_app"
        android:textColor="@color/black"
        android:textSize="@dimen/set_text_size"
        app:drawableEndCompat="@drawable/share" />
```

ilshat-abdulin commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,113 @@
+        android:layout_height="@dimen/zero"
+        android:layout_weight="2">
+
+    </LinearLayout>
```
Удали

AlexanderKorytin committed:
1. Изменен обработчик кнопки back в SettingsActivity.kt

2. Все иконки заменены на 24х24 с изменением отступов

3. В SettingsActivity.xml:
 - убрана излишняя вложенность
 - отступы верстки приведены в ссответствие с дизайном figma
 - <Button заменены на <TextView

 4. Цвет иконок SettingsActivity.xml внесен в colors.xml

AlexanderKorytin committed:
1. Изменен обработчик кнопки back в SettingsActivity.kt

2. Все иконки заменены на 24х24 с изменением отступов

3. В SettingsActivity.xml:
 - убрана излишняя вложенность
 - отступы верстки приведены в ссответствие с дизайном figma
 - <Button заменены на <TextView

 4. Цвет иконок SettingsActivity.xml внесен в colors.xml

AlexanderKorytin committed:
1. Добавлены иконки "назад" в меню "поиск" и "медиатека"

ilshat-abdulin commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,88 @@
+            android:src="@drawable/back"
+            tools:ignore="RtlSymmetry" />
+
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center"
+            android:layout_marginStart="24dp"
+            android:layout_marginTop="14dp"
+            android:layout_marginBottom="@dimen/padding_16"
+            android:text="@string/bot_settings"
+            android:textSize="@dimen/primary_text_size" />
```
В прошлый раз забыл осветить этот момент. Цвет текста должен быть черный

ilshat-abdulin commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,88 @@
+        android:foreground="?selectableItemBackground"
+        android:layout_height="wrap_content"
+        android:backgroundTint="@color/set_background"
+        android:paddingVertical="21dp"
+        android:paddingStart="@dimen/padding_16"
+        android:paddingEnd="12dp"
```
И про вынос повторяющихся размеров в ресурсы также не забывай.

```
android:paddingVertical="21dp"
android:paddingEnd="12dp"
```

AlexanderKorytin committed:
1. размеры 21 и 12 dp внесены в ресурсы
2. цвет текста в меню Setting, Find, Media установлен black

AlexanderKorytin committed:
1. Созданы стили для кнопок главного меню и текстовых полей меню настррики
2. Созданы стили для контенеров LinearLyout и текстовых заголовков
3. цветовые решения реализованны через темы для Main и Settings
4. реализован ночной режим для Main и Settings
